using/src/activations.rs

```diff
@@ -0,0 +1,178 @@
+//! Methods to select which of a set of implementations to activate based on a schema's
+//! `Request`s.
+
+use crate::{Find, Found, Implementations, Request, Schema};
+
+/// An Activation references a request and contains the min and max versioned implementations
+/// available in the implementation registry which can satisfy that request, or None if no
+/// matching implementations were available.
+// type Activation<'schema, 'impls, T> = (&'schema Request, Find<'impls, T>);
+
+impl<'a> Schema<'a> {
+    /// Given a set of implementations, take all requests in the document and
+    /// match them to the implementations which satisfy them ("satisfying
+    /// implementations").
+    ///
+    /// This returns an `Iterator` over `(&Request, Find<T>)`.
+    /// `Find` is an `Iterator` over `(&Version, &T)`, which will iterate satisfying
+    /// implementations ordered by lowest to highest version. The
+    /// [`bound`](Bounds.html#bound) method can be used to get the lowest and highest
+    /// satisfying versions.
+    ///
+    /// # Example:
+    ///
+    /// ```
+    /// use using::*;
+    ///
+    /// let impls = Implementations::new()
+    ///     .provide("https://spec.example.com/A", Version(1, 2), "impl for A 1.2")
+    ///     .provide("https://spec.example.com/A", Version(1, 3), "impl for A 1.3")
+    ///     .provide("https://spec.example.com/A", Version(1, 7), "impl for A 1.7");
+    /// let schema = Schema::parse(r#"schema @using(spec: "https://spec.example.com/A/v1.0") { query: Query }"#)?;
+    ///
+    /// let max = schema
+    ///   .activations(&impls)
+    ///   .filter_map(|(_request, mut impls)| impls.bounds())
+    ///   .collect::<Vec<_>>();
+    ///
+    /// assert_eq!(max, vec![(
+    ///     (&Version(1, 2), &"impl for A 1.2"), (&Version(1, 7), &"impl for A 1.7")
+    /// )]);
+    /// # Ok::<(), GraphQLParseError>(())
+    /// ```
+    pub fn activations<'impls, T>(
+        &'impls self,
+        implementations: &'impls Implementations<T>,
+    ) -> impl Iterator<
+        Item = (
+            &'a Request,
+            Find<'impls, T, impl Iterator<Item = Found<'impls, T>>>,
+        ),
+    > + 'impls
+    where
+        'impls: 'a,
+        T: 'impls,
+    {
+        self.using
```

Comment:
It's a field on `Schema`, which was defined elsewhere, but we've since moved everything in activations over to schema.rs, so this confusion shouldn't be an issue anymore